website/docs/configuration.md

```diff
@@ -3,8 +3,100 @@ id: configuration
+
+## What goes into `docusaurus.config.js`?
+
+You should not have to write your `docusaurus.config.js` from scratch even if you are developing your site. All templates should come with a `docusaurus.config.js` at root that includes the necessary fields for the initial site.
+
+However, it can be helpful if you have a high-level understanding of how the configurations are designed and implemented.
+
+For reference to each of the configurable fields, you may refer to the API reference of [docusaurus.config.js](/docs/docusaurus.config.js).
+
+The configurations can be categorized into:
+
+- [Site meta](#site-meta)
+- [Deployment configurations](#deployment-configurations)
+- [Theme configurations, plugins, and presets](#theme-plugins-and-presets-configurations)
+- [Custom configurations](#custom-configurations)
+
+### Site meta
+
+Site meta contains the essential meta information such as titles and `favicon`.
+
+They are used by your site app in a number of places such as your site's title and headings, browser tab icon, and SEO.
+
+- [title](/docs/docusaurus.config.js#title)
+- [tagline](/docs/docusaurus.config.js#tagline)
+- [favicon](/docs/docusaurus.config.js#favicon)
+
+### Deployment configurations
+
+Deployment configurations are used when you deploy your site with Docusaurus' deploy command. The related fields are:
+
+<!-- TODO: if we use monospace for the field names, they no longer look like a link -->
+
+<!-- TODO: currently these fields are only used in GH Pages, what about other deployment services such as Netlify -->
+
+- [url](/docs/docusaurus.config.js#url)
+- [baseUrl](/docs/docusaurus.config.js#baseUrl)
+- [organizationName](/docs/docusaurus.config.js#organizationname)
+- [projectName](/docs/docusaurus.config.js#projectName)
+
+You may also check the doc for [Deployment](/docs/deployment) for more information about the fields.
+
+### Theme, plugins, and presets configurations
+
+<!-- TODO: More explanation from these docs, respectively -->
+
+- [themeConfig](/docs/docusaurus.config.js#themeConfig)
+- [presets](/docs/docusaurus.config.js#presets)
+- [plugins](/docs/docusaurus.config.js#plugins)
+
+### Custom configurations
+
+You may have your own custom fields. And `docusaurus.config.js` will be aware of the fields and guard your configuration from unknown fields.
+
+To add a custom field, add the field name to `customFields`. Then, you may use the field for your customization data:
+
+```js
+// docusaurus.config.js
+module.exports = {
+  customFields: ['seo'],
+  seo: {
+    image: '',
+    keywords: [],
+  },
+};
+```
+
+- [customFields](/docs/docusaurus.config.js#customFields)
+
+## Accessing configuration from your site
+
+Your configuration object will be made available to all the components of your site. And you may access them via context as `siteConfig`:
+
+```jsx
+import React from 'react';
+import Helmet from 'react-helmet';
```

Comment:
Replace it with docusaurus/Head

There are various design reason for this. For example why do we use docusaurus/Link instead of react-router link ? Because if we switch to reach router or other router, this wont break user.

Similar case for helmet. React helmet is pretty much unmaintained now, there is a possibility we might switch. So docusaurus/Head is a better abstraction